### PR TITLE
Adds automatic highlighting of syndicate codewords for traitors and !!!GHOSTS!!!

### PR DESCRIPTION
**Pull request by silicons**
Syndicate codewords are highlighted in SLIGHTLY LARGER BLUE WAVY CURSIVE CLUWNE FONT.
Syndicate code responses are highlighted in the same way, but green instead!
Only other traitors, and ghosts can see this.
Ghosts being able to see this might lead to metagame problems, but uh, feedback would be nice.
Removes the excuse of "I DIDNT READ MY CODEWORDS"
Adds teamwork. Maybe too much of it.
IF THERE IS ANY OTHER ANTAG TYPE THAT GETS CODEWORDS PLEASE LET ME KNOW SO I CAN ADD THEM TO THE LIST.
I originally wanted to add this to PDA messages but I think that would be too much effort. Besides, it's a PDA message, you should have plenty of time (and actually think) to scan it for codewords.
Oh, and I guess traitors can rat each other out easier now, but they could do it anyways!
:cl: Mekhi Anderson
rscadd: Syndicate Agents who are provided with codewords are now mentally conditioned to instantly recognize them during normal vocal conversations, after complaints that agents either didn't bother or did not think of analyzing their conversations for these codewords.
tweak: Syndicate code words and responses will be highlighted cursive-blue or green, respectively, to Traitors.
experimental: Ghosts can also recognize codewords/responses in chat as autohighlighted.
fix: Syndicate Command apologizes for spelling "Beepsky Smash" "BeepKSY Smash".
:cl:

### CODE REVIEW

File: data/mode.txt [hunk 1]
```diff
@@ -1 +1 @@
-extended
```
Comment (kevinz000) [inline data/mode.txt]: wha- FUCK

File: code/__HELPERS/names.dm [hunk 2]
```diff
@@ -170,7 +170,7 @@ var/syndicate_code_response//Code response for traitors.
...
 	var/nouns[] = list("love","hate","anger","peace","pride","sympathy","bravery","loyalty","honesty","integrity","compassion","charity","success","courage","deceit","skill","beauty","brilliance","pain","misery","beliefs","dreams","justice","truth","faith","liberty","knowledge","thought","information","culture","trust","dedication","progress","education","hospitality","leisure","trouble","friendships", "relaxation")
-	var/drinks[] = list("vodka and tonic","gin fizz","bahama mama","manhattan","black Russian","whiskey soda","long island tea","margarita","Irish coffee"," manly dwarf","Irish cream","doctor's delight","Beepksy Smash","tequila sunrise","brave bull","gargle blaster","bloody mary","whiskey cola","white Russian","vodka martini","martini","Cuba libre","kahlua","vodka","wine","moonshine")
+	var/drinks[] = list("vodka and tonic","gin fizz","bahama mama","manhattan","black Russian","whiskey soda","long island tea","margarita","Irish coffee"," manly dwarf","Irish cream","doctor's delight","Beepsky Smash","tequila sunrise","brave bull","gargle blaster","bloody mary","whiskey cola","white Russian","vodka martini","martini","Cuba libre","kahlua","vodka","wine","moonshine")
```
Comment (optimumtact) [inline code/__HELPERS/names.dm]: sick powergaming buff

Comment (kevinz000) [inline code/__HELPERS/names.dm]: >:(

**Comment by kevinz000** [comment]
going to fix everything tomorrow.

**Comment by Giacom** [comment]
Do traitors still have a chance of only getting phrases or responses?

File: code/__HELPERS/names.dm [hunk 3]
```diff
@@ -217,9 +215,36 @@ var/syndicate_code_response//Code response for traitors.
...
+	var/parse = text
+	for(var/word in syndicate_code_phrase)
+		var/indexl = 0
+		var/indexu = 0
+		var/parse1
+		var/parse2
+		var/insertword = "<span class='codephrase'>[word]</span>"
+		if(!findtext(parse, word))
+			continue
+		indexl = findtext(parse, word)
```
Comment (RemieRichards) [inline code/__HELPERS/names.dm]: Fix this ^, it's minor but I'll kill you.

File: code/__HELPERS/names.dm (hunk 2 above)
Comment (RemieRichards) [inline code/__HELPERS/names.dm]: can you fix these lists.
```var/blah[]``` -> ```var/list/blah```

File: code/__HELPERS/names.dm (hunk 3 above)
Comment (RemieRichards) [inline code/__HELPERS/names.dm]: Actually know this proc is bad.
replace it with:
```
/proc/parse_for_code_words(text)
    for(var/word in syndicate_code_phrase)
        text = replacetext(text, word, "<span class='codephrase'>[word]</span>")
    for(var/word in syndicate_code_response)
        text = replacetext(text, word, "<span class='coderesponse'>[word]</span>")
   return text
```

Comment (Giacom) [inline code/__HELPERS/names.dm]: Though you'll end up with this:

"I didn't **CAT**ch that." You'll want to add some sort of space separation.

Comment (RemieRichards) [inline code/__HELPERS/names.dm]: Yep, true, you can just swap ```replacetext(text, word, blah)``` for ```replacetext(text, " [word] ", blah)``` though.

Comment (AnturK) [inline code/__HELPERS/names.dm]: "Look at my cat."

I'd say using \b would be better but this is pretty performance sensitive, and i'm not sure how byond regex efficiency goes.

**Comment by KorPhaeron** [comment]
>Adds teamwork

But we already have a thousand team modes.

And yes @Giacom raises a good point, traitors often only get some of the words.

**Comment by Supermichael777** [comment]
it should highlight on words said by non traitors. because bar, clown, Triple citrus isnt going to get annoying

**Comment by Iamgoofball** [comment]
@korphaeron @giacom We fixed that ages ago

**Comment by lzimann** [comment]
I don't think we need even more hivemind-like features.

**Comment by XDTM** [comment]
It's not like they couldn't do it before, plus it's not private like hivemind and if you drop codewords randomly you'll get valided.

**Comment by Durkel** [comment]
It's really okay to not hold the players hand 24/7. I think we can manage with players having to actually look at their notes or actually remember something.

**Comment by CyberbossJHCB** [comment]
@Durkel I make a point of saying some code words near the start of each game, no one has EVER responded

**Comment by KorPhaeron** [comment]
Guess you gotta play your solo antag solo then

**Comment by TechnoAlchemisto** [comment]
Basically what Durkel said, traitor is a solo antag with optional teamwork.

Although this makes sense, it really is too convenient and weird. You're making it too easy, if people really want to work as a team they'll buy the encryption key anyways, no point in randomly highlighting words that might not even be codewords.

**Comment by ChangelingRain** [comment]
It should probably only highlight words the traitor has as codewords/responses, yeah.

**Comment by TechnoAlchemisto** [comment]
Even then, a lot of these things are common words which is why they're 'inconspicious' context is very important when determining if someone is actually using the codewords for identification. Highlighting them regardless of context is very hand-holdey and will more often than not be wrong.

**Comment by kevinz000** [comment]
It's up to them to determine if it's a false positive.

Also I'm considering making codewords a var .
@ChangelingRain  traitors have all codewords tho
Also codewords can be recognized even if it's meta-y to do so ....

**Comment by Mysak0CZ** [comment]
It is good idea, but player shoud hale last word, if he uses it - via option in game preferences

**Comment by ChangelingRain** [comment]
Traitors don't have all codewords, each traitor only gets like three of each if that!

**Comment by kevinz000** [comment]
um isn't only 3 generated?

**Comment by lzimann** [comment]
Apparently they all get the same now. Same codes and same responses.

**Comment by Militaires** [comment]
it was a part of the deception to think that someone is dropping codewords when they're just saying normal things, now this is eh and kind of makes syndi key have no real point.

hand-holdey.

**Comment by MrPerson** [comment]
You know what TechnoAlchemist? Go fuck yourself. "This is too convenient" It's a fucking game you mook

**Comment by TechnoAlchemisto** [comment]
Big glowing words make the game even more brainless, convenience often limits player involvement with content. 

Also, please dont act like that.

**Comment by kevinz000** [comment]
@Militaires The deception still works, traitors knew the codewords anyways so nontraitors never could trick them into believing they know codes.
It's just now they can notice the same things easier without having to keep notes open.

**Comment by KorPhaeron** [comment]
It's a "MrPerson thinks games should be automated to improve quality of life" episode

**Comment by optimumtact** [comment]
@KorPhaeron @TechnoAlchemisto 

Your position on this is rediculous, in all of the rounds I have played I have never ever once managed to coordinate with another traitor via codewords. This will increase the likelihood of that event happening which will improve gameplay for traitors and for crew.

Nothing about this should be even slightly controversial, it's taking something that is already supposed to happen and making it not completely useless.

**Comment by TechnoAlchemisto** [comment]
If you're going to force this, make it a pref. I don't want randomly highlighted words when I am playing traitor, it's dumb. If you want to use codewords you will remember them, if you don't, you won't. 

Although I can count on one hand how many times I have used codewords to coordinate with another traitor, this is mostly because coordinating with other traitors is generally a good way to get killed.

 If I want to coordinate with others I will buy a encryption key.

**Comment by KorPhaeron** [comment]
How will turning traitor into nuke ops lite "improve gameplay for the crew"

**Comment by kevinz000** [comment]
yes it will have a pref

**Comment by MrPerson** [comment]
If you don't want the codewords at all, ask for them to be removed outright.

But since we have them, it would be better if they weren't completely useless

**Comment by PKPenguin321** [comment]
I'd just like to say I've coordinated through code words via radio, PDA, and person to person chatting

**Comment by PKPenguin321** [comment]
That was sent prematurely it was in response to this
>in all of the rounds I have played I have never ever once managed to coordinate with another traitor via codewords.

I think this will be nice 99% of the time, but code words being highlighted on the radio might be a little *too* good, I don't know

**Comment by KorPhaeron** [comment]
I'd rather codewords be removed than this be merged

**Comment by kevinz000** [comment]
rude as fuck

**Comment by scoopscoop** [comment]
I think I've only ever seen codewords used properly once. Since it's been cheapened I see the encryption key used a lot more than codewords, but perhaps this would make traitors cooperate a bit too much even with the possibility of betrayal hanging in the air.

Perhaps the formatting should only happen when hearing someone directly?

**Comment by optimumtact** [comment]
may as well just give traitors a syndicate radio key on spawn.

**Comment by KorPhaeron** [comment]
every other antag has a magic hivemind comm channel, may as well at this point

**Comment by Iamgoofball** [comment]
I'm reopening this

**Comment by CyberbossJHCB** [comment]
do eet

**Comment by KorPhaeron** [comment]
@Iamgoofball 

Why not just make a "team infiltration" mode variant where all the traitors get a free syndie key if you want that so badly

**Comment by Iamgoofball** [comment]
could be good actually

**Pull request closed without merging**